src/Application.vala

```diff
@@ -216,15 +220,15 @@ public class Terminal.Application : Gtk.Application {
     protected override int command_line (ApplicationCommandLine command_line) {
         unowned var options = command_line.get_options_dict ();
         var window = (MainWindow) active_window;
-        bool new_window;
+        bool new_window = false;
+        options.lookup ("new-window", "b", out new_window);
+        /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
+         * Currently they are set to restore (subject to the restore-tabs setting).
+         * If it is desired that tabs should never be restored in these circimstances add another check below.
+         */
 
-        if (window == null || options.lookup ("new-window", "b", out new_window) && new_window) {
-            /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
-             * Currently they are set to restore (subject to the restore-tabs setting).
-             * If it is desired that tabs should never be restored in these circimstances add another check below.
-             */
-            bool restore_tabs = !("commandline" in options || "execute" in options) || window == null;
-            window = new MainWindow (this, restore_tabs);
+        if (window == null || new_window) {
```

Comment:
keep the options.lookup() inline here, it's not only getting the value of `new-window` option, but validating that it's a boolean value too.

Reply:
That will not work.  We need a value for `new_window` when `window` is `null`.  If `window` is `null` and the lookup is inline then `new_window` will be undefined because of lazy evaluation of OR expression.

Reply:
you can use `window != null` in the restore_tabs argument, that has the same meaning of `new_window` being true.